dnserver/main.py

```diff
@@ -173,3 +181,6 @@ def stop(self):
     @property
     def is_running(self):
         return (self.udp_server and self.udp_server.isAlive()) or (self.tcp_server and self.tcp_server.isAlive())
+
+    def add_record(self, zone: Zone):
+        self.records.zones.append(zone)
```

Comment:
humm, have you actually tried this? I think because of `self.records = [Record(zone) for zone in records.zones]` this won't actually add another record to the dns server.

Best option would be to store `Records` directly on `ProxyResolver` and `BaseResolver`, with that, extending `self.records` should actually affect the dns server.

Reply:
> humm, have you actually tried this? I think because of self.records = [Record(zone) for zone in records.zones] this won't actually add another record to the dns server.

Well, there is a unit test for that. Do you think this is not tested enough?

> Best option would be to store Records directly on ProxyResolver and BaseResolver, with that, extending self.records should actually affect the dns server.

I just pushed that. Indeed, doing this way actually not requires the server to be restarted when a zone is added.